tasks/config-gitlab-runner.yml

```diff
@@ -1,37 +1,62 @@
 ---
-- name: Set advanced configuration | global section
+- name: Set advanced configuration | global section values
   lineinfile:
     dest: /etc/gitlab-runner/config.toml
-    regexp: '^(.*){{ item.param }} ='
-    line: '\1{{ item.param }} = {{ item.value }}'
+    regexp: '^(.*){{ item.key }} ='
+    line: '\1{{ item.key }} = {{ item.value }}'
     state: present
     backrefs: true
   notify: restart gitlab-runner
-  loop:
-    - param: concurrent
-      value: '{{ global_section.concurrent }}'
-  vars:
-    global_section: '{{ gitlab_runner_config.global }}'
+  with_dict: "{{ gitlab_runner_config.global_values | default({}) }}"
```

Comment:
Why with_dict instead of loop?

Reply:
We could use `loop` but in that case we've to add the filter *dict2items* at the end, but I think using *with_dict* is more self explanatory.

Reply:
> We could use `loop` but in that case we've to add the filter _dict2items_ at the end, but I think using _with_dict_ is more self explanatory.

Ansible recommend use loop for most use cases.

Reply:
> > We could use `loop` but in that case we've to add the filter _dict2items_ at the end, but I think using _with_dict_ is more self explanatory.
> 
> Ansible recommend use loop for most use cases.

My bad, I didn't read the suggestion of moving to *loop* from *with_X*.
Going to implement it